clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -260,12 +262,14 @@ public void handshake() throws IOException {
             maybeThrowSslAuthenticationException();
 
             // this exception could be due to a write. If there is data available to unwrap,
-            // process the data so that any SSLExceptions are reported
+            // process the data so that any SSL handshake exceptions are reported
```

Comment:
It would be good to elaborate on why we need to do this as it's not obvious by just reading the code.

Reply:
Added comments in a new method to process the exceptions.